ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+//! Bindings for [`AFont`], [`AFontMatcher`], and [`ASystemFontIterator`]
+//!
+//! [`AFont`]: https://developer.android.com/ndk/reference/group/font#afont_close
+//! [`AFontMatcher`]: https://developer.android.com/ndk/reference/group/font#afontmatcher_create
+//! [`ASystemFontIterator`]: https://developer.android.com/ndk/reference/group/font#asystemfontiterator_open
+
+use std::convert::TryFrom;
+use std::ffi::{CStr, OsStr};
+use std::fmt;
+use std::os::unix::prelude::OsStrExt;
+use std::path::Path;
+use std::ptr::NonNull;
+
+/// An integer holding a valid font weight value between 1 and 1000.
+///
+/// See the followings for more details:
+/// * [`AFONT_WEIGHT_*`]
+/// * [`Font::weight`]
+///
+/// [`AFONT_WEIGHT_*`]: https://developer.android.com/ndk/reference/group/font#anonymous-enum-33
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct FontWeight(u16);
+
+impl FontWeight {
+    pub const fn new(value: u16) -> std::result::Result<Self, TryFromU16Error> {
+        if 0 < value && value <= 1000 {
+            Ok(Self(value))
+        } else {
+            Err(TryFromU16Error(()))
+        }
+    }
+
+    pub const fn value(&self) -> u16 {
+        self.0
+    }
+
+    pub const THIN: FontWeight = FontWeight(ffi::AFONT_WEIGHT_THIN as u16);
+    pub const EXTRA_LIGHT: FontWeight = FontWeight(ffi::AFONT_WEIGHT_EXTRA_LIGHT as u16);
+    pub const LIGHT: FontWeight = FontWeight(ffi::AFONT_WEIGHT_LIGHT as u16);
+    pub const NORMAL: FontWeight = FontWeight(ffi::AFONT_WEIGHT_NORMAL as u16);
+    pub const MEDIUM: FontWeight = FontWeight(ffi::AFONT_WEIGHT_MEDIUM as u16);
+    pub const SEMI_BOLD: FontWeight = FontWeight(ffi::AFONT_WEIGHT_SEMI_BOLD as u16);
+    pub const BOLD: FontWeight = FontWeight(ffi::AFONT_WEIGHT_BOLD as u16);
+    pub const EXTRA_BOLD: FontWeight = FontWeight(ffi::AFONT_WEIGHT_EXTRA_BOLD as u16);
+    pub const BLACK: FontWeight = FontWeight(ffi::AFONT_WEIGHT_BLACK as u16);
+    pub const MAX: FontWeight = FontWeight(ffi::AFONT_WEIGHT_MAX as u16);
```

Comment:
`AFONT_WEIGHT_MIN` exists in the bindings too? Either have both or drop both :)

Reply:
I think both `MIN` and `MAX` are needed. The problem is, `AFONT_WEIGHT_MIN` is 0, but the OpenType spec says a font weight should be greater than 0. Considering the case like [std::num::NonZeroUsize](https://doc.rust-lang.org/stable/std/num/struct.NonZeroUsize.html#associatedconstant.MIN), `FontWeight::MIN` should be `FontWeight(1)`, but that's not consistent with other bindings.

Reply:
I edited that part as follows, how about this?
```rust
    pub const fn new(value: u16) -> Result<Self, TryFromU16Error> {
        if Self::MIN.0 <= value && value <= Self::MAX.0 {
        ...
    }

    // + 1 is to keep consistency between FontWeight and other types like std::num::NonZero*
    pub const MIN: FontWeight = FontWeight(ffi::AFONT_WEIGHT_MIN as u16 + 1);
```

Reply:
So `MIN` is exclusive, `MAX` is inclusive... We could just use a Rust `Range` to be more clear about the intent (may or may not be public):

```rust
    const VALID_RANGE: RangeInclusive<u16> =
        ffi::AFONT_WEIGHT_MIN as u16 + 1..=ffi::AFONT_WEIGHT_MAX as u16;
```

In favour of `MIN` and `MAX`?

Reply:
There are many custom numeric types like `NonZero*` or `rust_decimal::Decimal` in Rust, and AFAIK, such types always define inclusive `MIN` and `MAX`, as standard types such as `i32` also do so. I think `MIN` and `MAX` is sufficient for this.